README.md

```diff
@@ -140,9 +140,135 @@ export function App() {
 }
 ```
 
+#### TypeScript Generics
+
+We also support the use of Generics to add type definitions for your props. If
+used, the props will be properly typed when the props are passed to the
+`updateWithProps` method.
+
+To utilise generics you can use one of two methods. In both of the exampled
+below, we create a custom internal type called `MySketchProps` which is a union
+type of `SketchProps` and a custom type which has a `rotation` key applied to
+it.
+
+> Sidenote:
+>
+> We could also write the `MySketchProps` type as an interface to do exactly the
+> same thing if that is to your personal preference:
+>
+> ```typescript
+> interface MySketchProps extends SketchProps {
+>   rotation: number;
+> }
+> ```
+
+This means, in these examples, that when the `rotation` prop that is provided as
+part of the `props` passed to the `updateWithProps` function, it will be
+correctly typed as a `number`.
```

Comment:
I'm not sure if these lines are actually required, what do you think @yevdyko?

Reply:
@jamesrweb I think this makes sense as a more explicit explanation of the example. Thanks!